samples/Ark.ReferenceProject/Core/Ark.Reference.Core.Tests/Features/BookPrintProcessSteps.cs

```diff
@@ -0,0 +1,160 @@
+{
+    [Binding]
+    public sealed class BookPrintProcessSteps
+    {
+        private readonly TestClient _client;
+        private readonly TestHost _testHost;
+        private readonly string _controllerName = "bookprintprocess";
+        private int? _currentBookId;
+        private int? _currentPrintProcessId;
+        private BookPrintProcess.V1.Output? _currentPrintProcess;
+
+        public BookPrintProcessSteps(TestClient client, TestHost testHost)
+        {
+            _client = client;
+            _testHost = testHost;
+        }
+
+        [Given(@"I have created a book with title ""(.*)"" and author ""(.*)""")]
+        public void GivenIHaveCreatedABookWithTitleAndAuthor(string title, string author)
+        {
+            var request = new Book.V1.Create
+            {
+                Title = title,
+                Author = author,
+                Genre = BookGenre.Fiction,
+                ISBN = "978-0-123456-78-9"
+            };
+
+            _client.PostAsJson("book", request);
+            _client.ThenTheRequestSucceded();
+            
+            var result = _client.ReadAs<Book.V1.Output>();
+            _currentBookId = result.Id;
+        }
+
+        [Given(@"I have created a book print process for that book")]
+        [When(@"I create a book print process for that book")]
+        public void WhenICreateABookPrintProcessForThatBook()
+        {
+            var request = new BookPrintProcess.V1.Create
+            {
+                BookId = _currentBookId!.Value,
+                ShouldFail = false
+            };
+
+            _client.PostAsJson(_controllerName, request);
+        }
+
+        [Given(@"I have created a book print process for that book with ShouldFail (.*)")]
+        public void GivenIHaveCreatedABookPrintProcessWithShouldFail(bool shouldFail)
+        {
+            var request = new BookPrintProcess.V1.Create
+            {
+                BookId = _currentBookId!.Value,
+                ShouldFail = shouldFail
+            };
+
+            _client.PostAsJson(_controllerName, request);
+            _client.ThenTheRequestSucceded();
+                
+            var response = _client.ReadAs<BookPrintProcess.V1.Output>();
+            _currentPrintProcessId = response.BookPrintProcessId;
+        }
+
+        [When(@"I try to create another book print process for that book")]
+        public void WhenITryToCreateAnotherBookPrintProcessForThatBook()
+        {
+            var request = new BookPrintProcess.V1.Create
+            {
+                BookId = _currentBookId!.Value,
+                ShouldFail = false
+            };
+
+            _client.PostAsJson(_controllerName, request);
+        }
+
+        [When(@"I retrieve the print process status")]
+        public void WhenIRetrieveThePrintProcessStatus()
+        {
+            _client.Get($"{_controllerName}/{_currentPrintProcessId}");
+        }
+
+        [When(@"I wait background bus to idle and outbox to be empty")]
```

Comment:
add this attribut to TestHost existing Verb if the When variation is not present. Do not alias it here as is not specific to 'PrintingProcess' steps